stdlib/Dates/src/types.jl

```diff
@@ -124,7 +124,7 @@ end
 """
 struct Time <: TimeType
     instant::Nanosecond
-    Time(instant::Nanosecond) = new(instant)
+    Time(instant::Nanosecond) = new(mod(instant, 86400000000000))
```

Comment:
Could be good to have a comment about this number or make it a constant with a name.